json.lua

```diff
@@ -64,32 +64,43 @@ local function encode_table(val, stack)
   if stack[val] then error("circular reference") end
 
   stack[val] = true
+  -- Check whether to treat as a array or object
+  local array = true
+  local length = 0
+	local nLen = 0
```

Comment:
Have you considered the worst-case this might create: What if I serialize `{[1e9] = true}`?

Reply:
Missed all these. Revisiting after a long time. Thanks for catching that. Had not considered it at all. Been using it, did not have issues. Fixed it in the next commit by preventing sparse arrays greater than 100 (arbitrarility) length getting through. Seems like other json modules also handle sparse arrays in a undefined way. 100 should be good for many cases I suppose.

Reply:
Yeah, it's a tricky issue, and behavior varies wildly. I believe ideally it should be left up to the user. Perhaps use something like a "density" factor as a threshold?

Reply:
I have added a variable called SPARSELIMIT in the module that the user can change after requiring the module to change it. Default set it to 100.